sdk/src/Core/Sampling/ServiceConnector.cs

```diff
@@ -53,61 +56,103 @@ public ServiceConnector(DaemonConfig daemonConfig, AmazonXRayClient xrayClient)
-            _xrayClient = xrayClient;
+            _xrayConfig = xrayConfig;
         }
 
-        private AmazonXRayClient CreateXRayClient()
+        private XRayConfig CreateXRayConfig()
         {
-            _config.ServiceURL = $"http://{_daemonConfig.TCPEndpoint.Address}:{_daemonConfig.TCPEndpoint.Port}";
-            return new AmazonXRayClient(_credentials,_config);
+            var config = new XRayConfig();
+            config.ServiceURL = $"http://{_daemonConfig.TCPEndpoint.Address}:{_daemonConfig.TCPEndpoint.Port}";
+            return config;
         }
 
         private void RefreshEndPoint()
         {
             var serviceUrlCandidate = $"http://{_daemonConfig.TCPEndpoint.Address}:{_daemonConfig.TCPEndpoint.Port}";
-         
-            if (serviceUrlCandidate.Equals(_xrayClient.Config.ServiceURL)) return; // endpoint do not need refreshing
 
-            _config.ServiceURL = serviceUrlCandidate;
-            _xrayClient = new AmazonXRayClient(_credentials, _config);
-            _logger.DebugFormat($"ServiceConnector Endpoint refreshed to: {_xrayClient.Config.ServiceURL}");
+            if (serviceUrlCandidate.Equals(_xrayConfig.ServiceURL)) return; // endpoint do not need refreshing
+
+            _xrayConfig.ServiceURL = serviceUrlCandidate;
+            _logger.DebugFormat($"ServiceConnector Endpoint refreshed to: {_xrayConfig.ServiceURL}");
         }
 
         /// <summary>
-        /// Use X-Ray client to get the sampling rules
-        /// from X-Ray service.The call is proxied and signed by X-Ray Daemon.
+        /// Get the sampling rules from X-Ray service.The call is proxied and signed by X-Ray Daemon.
         /// </summary>
         /// <returns></returns>
-        public async Task<GetSamplingRulesResponse>  GetSamplingRules()
+        public async Task<GetSamplingRulesResponse> GetSamplingRules()
         {
-            List<SamplingRule> newRules = new List<SamplingRule>();
-            GetSamplingRulesRequest request = new GetSamplingRulesRequest();
-
-            Task<Model.GetSamplingRulesResponse> responseTask;
+            Task<WebResponse> responseTask;
             lock (_xrayClientLock)
             {
                 RefreshEndPoint();
-                responseTask = _xrayClient.GetSamplingRulesAsync(request);
+                responseTask = GetSamplingInfoAsync(_xrayConfig.ServiceURL + "/GetSamplingRules");
             }
             var response = await responseTask;
 
-            foreach(var record in response.SamplingRuleRecords)
+            var responseContent = GetResponseContent(response);
+
+            List<SamplingRule> samplingRules = UnmarshallSamplingRuleResponse(responseContent);
+
+            GetSamplingRulesResponse result = new GetSamplingRulesResponse(samplingRules);
+            return result;
+        }
+
+        private Task<WebResponse> GetSamplingInfoAsync(string url, string content = null)
+        {
+            var request = (HttpWebRequest)WebRequest.Create(url);
```

Comment:
In .NET Core using HttpWebRequest is highly frowned on as it is close to a deprecated class and has clean up issues. The preferred way is to use System.Net.Http.HttpClient.

Reply:
Switched to `HttpClient`